Modules/Module_Processing/Module_ASL/xASL_wrp_RegisterASL.m

```diff
@@ -314,13 +314,19 @@
 end
 
 %% H. Here we create a temporary dummy ASL image of which the image contrast is curated, for registration only
-[PWI, ~, ~, ~, Control] = xASL_im_ASLSubtractionAveraging(x, x.P.Path_despiked_ASL4D);
-xASL_io_SaveNifti(x.P.Path_despiked_ASL4D, x.P.Path_mean_PWI_Clipped, PWI, 16, false);
+[PWI, ~, ~, xSubtracted, Control] = xASL_im_ASLSubtractionAveraging(x, x.P.Path_despiked_ASL4D);
+jsonFields.Initial_PLD = xSubtracted.Q.InitialPLD_PWI;
+jsonFields.LabelingDuration = xSubtracted.Q.LabelingDuration_PWI;
+jsonFields.EchoTime = xSubtracted.Q.EchoTime_PWI;
+xASL_io_SaveNifti(x.P.Path_despiked_ASL4D, x.P.Path_mean_PWI_Clipped, PWI, 16, false, [], 0, xASL_bids_parms2BIDS(jsonFields, [], 1));

```

Comment:
Isn't this json saving done in xASL_io_ASLSubtractionAveraging?

Reply:
in IO yes, but in IM no...

Reply:
yes, so we don't use xASL_im_ASLSubtractionAveraging here but we use xASL_io_ASLSubtractionAveraging instead right?

Reply:
You were right. Using the IO function a659dc33